community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/OnlineIndexUpdates.java

```diff
@@ -113,8 +115,10 @@ private void gatherUpdatesFor( long nodeId, NodeCommand nodeCommand, List<Proper
                 gatherUpdatesFromCommandsForNode( nodeId, nodeCommand, propertyCommands );
 
         NodeUpdates nodeUpdates = nodePropertyUpdate.build();
-        updateService.loadAdditionalProperties( nodeUpdates );
-        updates.add( nodeUpdates );
+        for ( IndexEntryUpdate<LabelSchemaDescriptor> update :  updateService.convertToIndexUpdates( nodeUpdates ) )
```

Comment:
In this case we need to materialize here, as the when the `iterator()` is called, the store might have changed so the ´convertToIndexUpdates´ might not yield the same result.